xdsl/irdl/irdl.py

```diff
@@ -560,6 +560,7 @@ class AttrSizedSegments(IRDLOption):
     """
 
     attribute_name: ClassVar[str]
+    as_property: bool = False
```

Comment:
As long as we switch this to default True once the PR is in and some time has passed, I am fine with this :+1:

Reply:
If it's dominantly used with as_property=True then, I'm for it too :+1:

Reply:
Good point, opened #1606 :slightly_smiling_face: